From: sezero <sezero@users.sourceforge.net>
Date: Sun, 4 Aug 2019 23:00:40 +0000 (+0300)
Subject: Do not assume that sys/param.h defines MIN and MAX
X-Git-Url: http://git.xiph.org/?p=flac.git;a=commitdiff_plain;h=4a43f2bd4d7e12b5d994f48e18cf014008d204ce

Do not assume that sys/param.h defines MIN and MAX

One such example is sys/param.h from MinGW.
---

diff --git a/src/libFLAC/include/private/macros.h b/src/libFLAC/include/private/macros.h
index 02eada45..3a8072ec 100644
--- a/src/libFLAC/include/private/macros.h
+++ b/src/libFLAC/include/private/macros.h
@@ -51,8 +51,10 @@
 /* Whatever other unix that has sys/param.h */
 #elif defined(HAVE_SYS_PARAM_H)
 #include <sys/param.h>
+#if defined(MIN) && defined(MAX)
 #define flac_max(a,b) MAX(a,b)
 #define flac_min(a,b) MIN(a,b)
+#endif
 
 /* Windows VS has them in stdlib.h.. XXX:Untested */
 #elif defined(_MSC_VER)
